packages/beacon-node/src/sync/utils/pendingBlocksTree.ts

```diff
@@ -62,7 +63,11 @@ export function getUnknownAndAncestorBlocks(blocks: Map<RootHex, PendingBlock>):
 
   for (const block of blocks.values()) {
     const parentHex = block.parentBlockRootHex;
-    if (block.status === PendingBlockStatus.pending && block.blockInput == null && parentHex == null) {
+    if (
+      block.status === PendingBlockStatus.pending &&
+      (block.blockInput == null || block.blockInput.type === BlockInputType.blobsPromise) &&
```

Comment:
this does not look right, it's originally for unknown block sync and we don't care `blobsPromise` type there
but if we don't make a change here we have to change `triggerUnknownBlockSearch`

well this seems to me it's not ideal to mix missing blobs to this whole `UnknownBlockSync` as it currently deals with the whole chain with blocks while the concern here is just missing blobs for a specific block

I'd go with a separate class to avoid introducing any regressions, the current logic here in UnknownBlockSync is already complex and it takes time to test/detect bugs in this class

Reply:
i don't follow, treatment of this is like unknown block only, there is a followup coming where if only blobs show up then it would still end here with a new blockinput type.

lets say block in processed queue was rejected since blocks didn't become available, then whenever the block gets pulled here, it will go into the processing queue as normal, along with any ancestors it need to pull. 

imo this is a very complete flow for an "unavailable" block 

(and in followup we will even start search if only some/all of blobs are available after some cutoff and block isn't , but it will come with new blockinput type)

Reply:
> this does not look right, it's originally for unknown block sync and we don't care blobsPromise type there
but if we don't make a change here we have to change triggerUnknownBlockSearch

took more time on this, this is correct since you handled in `UnknownBlockSync.downloadBlock()` to download full block vs blobs. The "unknown" seems confusing here, we may know the block but not the blob so please add more comments on this part